terraform/eks/main.tf

```diff
@@ -196,13 +242,15 @@ resource "kubernetes_deployment" "aoc_deployment" {
 
 # create service upon the mocked server
 resource "kubernetes_service" "mocked_server_service" {
+  count = var.sample_app_mode == "pull" ? 0 : 1
```

Comment:
i would suggest we keep mock server no matter if it's a pull mode, so that we can also switch to use mock server to validate in the future.